### ISSUE TITLE
FATAL:  could not create lock file "/tmp/.s.PGSQL.5432.lock": Operation not permitted

### ISSUE THREAD

**Issue by bileser** [PostgresApp/PostgresApp#343]
Hi,

Postgress.app was running fantastically until today. When I started the App the following error log appeared:

**FATAL:  could not create lock file "/tmp/.s.PGSQL.5432.lock": Operation not permitted**

I am running Mac Os El Capitan 10.11.6 and ran the command

**sudo chmod 1777 /var/tmp**

(/tmp sym links to /var/tmp) . Permissions should be fine. What could be the issue?

Thx!

**Comment by jakob** [comment]
Maybe the lock file already exists? Try the following:

1) Quit Postgres.app
2) Open Activity Monitor, ensure that no process named 'postgres' is still running (if there are any, kill them starting with the one with lowest PID)
3) Type `ls -a /tmp` to see if the socket file and lock file are still there (`/tmp/.s.PGSQL.5432` and `/tmp/.s.PGSQL.5432.lock`). If they are, delete them.
4) restart Postgres.app

**Comment by bileser** [comment]
thx, jakob,
unfortunately I did 1) and 2) and 3) but no files were in /tmp.
The problem is, that the lock file according to the error log can't be _created_, although it should.
Thats my private dir, where tmp is located what /tmp is sym linking to:
drwxr-xr-x  90 root  wheel  3060  4 Okt 13:54 etc
drwxr-xr-x   2 root  wheel    68  1 Aug  2015 tftpboot
drwxrwxrwt@  3 root  wheel   102  4 Okt 13:36 tmp
drwxr-xr-x  24 root  wheel   816  4 Okt 10:41 var

**Comment by bileser** [comment]
Yup, was a Mac error. Did [this](http://stackoverflow.com/questions/32659348/operation-not-permitted-when-on-root-el-capitan-rootless-disabled) and was able to run sudo chmod -R 1777 /private/tmp successfully. Now every program can write files. But why do I have to do that??

**Comment by jakob** [comment]
@bileser This is strange. I don't know why your /tmp dir was borked. Every app should be able to write files to /tmp; that's the whole point of /tmp.

I would recommend re-enabling SIP if everything works now.

(BTW: Another possible workaround would have been to change the socket location to a different directory than /tmp in postgresql.conf)

**Comment by bileser** [comment]
thx, jakob,

yes, it is a really strange mac os problem. Another program not able to write in the /tmp directory lead me to the problem. If I re-enable SIP, Mac OS forgets my chmod.
Anyway, this is another problem and has nothing to do with your fantastic software, which works great!

**Comment by arakawarx7** [comment]
I was just experiencing this issue. My solution was to follow @bileser tip by disabling the SIP via `$ csrutil disable` through the OSX recovery tool.

After a reboot is when I started experiencing this issue. Weird 🐸

**Comment by jakob** [comment]
Hey everyone!

Generally speaking, disabling System Integrity Protection is not a good idea -- just like you wouldn't fix a blown fuse by shorting it with a nail. So make sure to reenable SIP after changing the broken permissions.

Considering that SIP is supposed to prevent exactly this type of problem, it looks like there must be a bug in macOS. It would be really great if you could report this problem at bugreport.apple.com

**Comment by chriszrc** [comment]
Oddly this happened to me today as well after a reboot. It had been a long time since I rebooted, and I noticed my startup chime had come back on, maybe because of a security update or something. So I did run:

`sudo nvram StartupMute=%01`

After I rebooted again, I got the create lock file issue. I have no idea if they are related. Oddly, I tried resetting th pvram, and it didn't seem to bring the chime back, nor did it fix postgres. 

I tried to reset the permissions on the folder with:

`sudo chmod 1777 /private/tmp`

And that actually just worked. Hopefully this is helpful to someone, lol-
